### PR TITLE
Add a DRM Sysfs IOGroup

### LINKED ISSUE #3369: SYSFS driver for IOGroup exists to facilitate L0 access.

### PR DESCRIPTION
**Pull request by dannosliwcd**
- Resolves #3369

### CODE REVIEW

File: libgeopmd/Makefile.am [hunk 1]
```diff
@@ -103,20 +111,21 @@ libgeopmd_la_LDFLAGS = $(AM_LDFLAGS) -version-info $(geopm_abi_version)
...
                 # end
 
 # Sysfs JSON definitions. From each JSON file, generate a same-prefixed cpp file,
 # and use {driver}_sysfs_json as the variable identifier
 # E.g., src/sysfs_attributes_cpufreq.cpp is made from
 # src/sysfs_attributes_cpufreq.json and names the JSON variable cpufreq_sysfs_json.
 sysfs_cpp_files = \
-		  src/sysfs_attributes_cpufreq.cpp \
+		  $(top_builddir)/gen-src/sysfs_attributes_cpufreq.cpp \
+		  $(top_builddir)/gen-src/sysfs_attributes_drm.cpp \
```
Comment (bgeltz) [inline libgeopmd/Makefile.am]: whitespace - align under the \ from the previous line

File: libgeopmd/Makefile.am [hunk 2]
```diff
@@ -245,6 +256,9 @@ libgeopmd_la_SOURCES = $(include_HEADERS) \
                        src/geopm_sched.c \
                        src/geopm_shmem.cpp \
                        src/geopm_version.c \
+                       # end
+
+nodist_libgeopmd_la_SOURCES = \
                        $(msr_cpp_files) \
                        $(sysfs_cpp_files) \
                        # end
```
Comment (bgeltz) [inline libgeopmd/Makefile.am]: whitespace - align under the \ from the previous line

File: libgeopmd/Makefile.am [hunk 3]
```diff
@@ -307,6 +321,21 @@ else
...
+	$(top_srcdir)/src/json_data.cpp.in > $@
+
+$(sysfs_cpp_files): $(top_builddir)/gen-src/%.cpp: $(top_srcdir)/json_data/%.json $(top_srcdir)/src/json_data.cpp.in
+	sed -e '/@JSON_CONTENTS@/ {' \
+	    -e "r $<" \
+	    -e 'd' -e '}' \
+	    -e "s/@JSON_IDENTIFIER@/$(lastword $(subst _, ,$*))_sysfs_json/g" \
+	$(top_srcdir)/src/json_data.cpp.in > $@
+
+
```
Comment (bgeltz) [inline libgeopmd/Makefile.am]:
```suggestion
```

File: libgeopmd/src/DrmSysfsDriver.cpp [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+        : DrmSysfsDriver(platform_topo(), DRM_DIRECTORY)
+    {
+    }
+
+    DrmSysfsDriver::DrmSysfsDriver(const PlatformTopo &topo,
+                                   const std::string &drm_directory)
+        : M_PROPERTIES{SysfsDriver::parse_properties_json(plugin_name(), drm_sysfs_json())}
+        , M_DRM_RESOURCE_BY_GPU_TILE(load_resources_by_gpu_tile(
+                    drm_directory,
+                    (topo.num_domain(GEOPM_DOMAIN_GPU) > 0) ? (topo.num_domain(GEOPM_DOMAIN_GPU_CHIP) / topo.num_domain(GEOPM_DOMAIN_GPU)) : 0))
```
Comment (bgeltz) [inline libgeopmd/src/DrmSysfsDriver.cpp]:
```suggestion
        , M_DRM_RESOURCE_BY_GPU_TILE(load_resources_by_gpu_tile(
                                     drm_directory,
                                     (topo.num_domain(GEOPM_DOMAIN_GPU) > 0) ? 
                                     (topo.num_domain(GEOPM_DOMAIN_GPU_CHIP) /
                                     topo.num_domain(GEOPM_DOMAIN_GPU)) : 0))
```
At least the drm_directory should be aligned under the load_resources... from the previous line.   Not sure if I broke your ternary expression or not.

**Review by bgeltz — COMMENTED** [review]
A few nits but otherwise pretty good!

File: docs/json_data/sysfs_attributes_drm.json [hunk 5]
```diff
@@ -0,0 +1,448 @@
...
+            "behavior": "variable",
+            "format": "double",
+            "units": "hertz",
+            "writeable": false,
+            "alias": ""
+        },
+        "BASE_ACT_FREQ": {
+            "attribute": "base_act_freq_mhz",
+            "scalar": 1e6,
+            "description": "Actual GPU base-doc_domain frequency selected by the power manager",
```
Comment (cmcantalupo) [inline docs/json_data/sysfs_attributes_drm.json]: fix find/replace error (GPU base frequency)

File: docs/source/geopm_pio_sysfs.7.rst [hunk 6]
```diff
@@ -58,6 +71,19 @@ to the ``cpufreq`` driver. For more information, see the
...
+to the ``i915 DRM`` driver.  For more information, see the
+`i915 documentation <https://www.kernel.org/doc/html/next/gpu/i915.html>`_ and the
+`oneAPI GPU Optimization Guide <https://www.intel.com/content/www/us/en/docs/oneapi/optimization-guide-gpu/2024-0/configuring-gpu-device.html>`_.
+The i915 driver is available in `upstream Linux <https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/gpu/drm/i915>`_.
+Additional features are available in the `out-of-tree version of the driver <https://github.com/intel-gpu/intel-gpu-i915-backports>`_. This
+IOGroup is intended for use with either version of the driver.
+
+.. geopm-sysfs-json:: DRM ../json_data/sysfs_attributes_drm.json
+   :no-signals:
+
```
Comment (cmcantalupo) [inline docs/source/geopm_pio_sysfs.7.rst]: Avoid copy paste maintenance

**Review by cmcantalupo — COMMENTED** [review]
copy pasta

**Review by cmcantalupo — APPROVED** [review]

**Review by bgeltz — APPROVED** [review]

**Pull request merged**